// angular-popper/dist/angular-popper.bundle.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('popper.js'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', 'popper.js', '@angular/common'], factory) :
	(factory((global['angular-icon'] = {}),global.core,global.Popper,global.common));
}(this, (function (exports,core,Popper,common) { 'use strict';

Popper = Popper && Popper.hasOwnProperty('default') ? Popper['default'] : Popper;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var PopperComponent = /** @class */ (function () {
    function PopperComponent(el, zone) {
        this.el = el;
        this.zone = zone;
        this.show = false;
        this.closeButton = true;
        this.placement = 'bottom';
        this.close = new core.EventEmitter();
    }
    PopperComponent.prototype.ngAfterViewInit = function () {
        this.create();
    };
    PopperComponent.prototype.ngOnChanges = function (changes) {
        if (changes.target && !changes.target.firstChange ||
            changes.placement && !changes.placement.firstChange) {
            this.destroy();
            this.create();
        }
    };
    PopperComponent.prototype.ngOnDestroy = function () {
        this.destroy();
    };
    PopperComponent.prototype.onClose = function () {
        this.show = false;
        this.close.emit();
    };
    PopperComponent.prototype.create = function () {
        var _this = this;
        this.zone.runOutsideAngular(function () {
            _this.popper = new Popper(_this.getTargetNode(), _this.el.nativeElement.querySelector('.angular-popper'), {
                placement: _this.placement
            });
        });
    };
    PopperComponent.prototype.destroy = function () {
        var _this = this;
        if (this.popper) {
            this.zone.runOutsideAngular(function () {
                _this.popper.destroy();
            });
            this.popper = null;
        }
    };
    PopperComponent.prototype.getTargetNode = function () {
        if (this.target) {
            if (typeof this.target === 'string') {
                return document.querySelector(this.target);
            }
            else {
                return this.target;
            }
        }
        else {
            return this.el.nativeElement;
        }
    };
    __decorate([
        core.Input(),
        __metadata("design:type", Object)
    ], PopperComponent.prototype, "show", void 0);
    __decorate([
        core.Input(),
        __metadata("design:type", Object)
    ], PopperComponent.prototype, "closeButton", void 0);
    __decorate([
        core.Input(),
        __metadata("design:type", String)
    ], PopperComponent.prototype, "placement", void 0);
    __decorate([
        core.Input(),
        __metadata("design:type", Object)
    ], PopperComponent.prototype, "target", void 0);
    __decorate([
        core.Output(),
        __metadata("design:type", Object)
    ], PopperComponent.prototype, "close", void 0);
    PopperComponent = __decorate([
        core.Component({
            selector: 'angular-popper',
            templateUrl: './angular-popper.component.html',
            styleUrls: ['./angular-popper.css'],
            changeDetection: core.ChangeDetectionStrategy.OnPush
        }),
        __metadata("design:paramtypes", [core.ElementRef, core.NgZone])
    ], PopperComponent);
    return PopperComponent;
}());

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var NgPopper = /** @class */ (function () {
    function NgPopper() {
    }
    NgPopper = __decorate$1([
        core.NgModule({
            imports: [
                common.CommonModule
            ],
            declarations: [
                PopperComponent
            ],
            exports: [
                PopperComponent
            ]
        })
    ], NgPopper);
    return NgPopper;
}());

exports.NgPopper = NgPopper;
exports.PopperComponent = PopperComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
